File: C++/0x03/3273.cpp
/*
이 문제는 단순하게 풀면 안됨
이중 for문으로 풀게 되면 시간 초과가 되게 됨. 너무 많이 돌아야해서.
그러므로 일반 for문에서 x에서 num[i]를 빼주고 그것이 true일때만 count가 가능
*/
#include <bits/stdc++.h>//자주 사용 라이브러리 전부 컴파일
using namespace std;//std 네임 스페이스를 사용
//두수의 합
int n,num[100001]={};
bool occur[2000001];
int x;
int main(void){
  int count=0;
  ios::sync_with_stdio(0);
  cin.tie(0);
  cin>>n;
  for(int i=0;i<n;i++) cin>>num[i];
  cin>>x;
  for(int i=0;i<n;i++){
    if(x-num[i]>0 && occur[x-num[i]]) count++;
    occur[num[i]]=true;
  }
  cout<<count;
}

File: Python/2022_3/3273.py
#이 문제도 이중 for문 불가. if문으로 풀어주어야 함.
#elif에서 j를 빼주는 이유는 크면 결국은 성립 안된다는 의미이기 때문이다.
n=int(input())
num=list(map(int,input().split()))
x=int(input())
count=0

num.sort()

i=0
j=len(num)-1
while i!=j:
    if(num[i]+num[j]==x):
        count+=1
        i+=1
    elif(num[i]+num[j]>x):
        j-=1
    else:i+=1
print(count)
